docs/getting-started.md

```diff
@@ -130,8 +180,10 @@ This schema allows querying for a list of shows, optionally filtering by title.
 ## Implement a Data Fetcher
 
 Data fetchers are responsible for returning data for a query.
+With the new Spring-GraphQL integration, it is technically possible to mix and match the DGS/Spring-GraphQL programming models. 
+However, to maintain consistency in your codebase and to take full advantage of DGS features, we recommend sticking with the DGS programming model.
```

Comment:
Where did we land on disabling the spring-graphql model by default? I'm still inclined to say it should be disabled by default, because it's hard for me to imagine why you would want to mix and match (with today's implementation) unless you were just migrating to plain spring-graphql. Otherwise, I suspect folks will just end up wondering why DGS features aren't working how they expect.

Reply:
So here's the current thinking (which is different from what I had in mind a week ago in light of recent  experimentation). It's a bit wordy,  but wanted to write everything down so we can discuss the longer term strategy as well.

The integration in it's current state  does technically allows folks to define spring-graphql data fetchers/data loaders, but does not guarantee all DGS features will work in the context of spring-graphql. Therefore, we will strongly encourage folks to stick with defining DGS style data fetchers and data loaders when using the framework.
However, if folks choose to still experiment or use it for some reason, it is very much expected that not everything will work.  It is unlikely that users will accidentally mix DGS and spring graphql style controllers following our docs. If they do, it is probably intentional knowing that only spring-graphql features can be expected.

In the future, it seems reasonable to expect that we will directly integrate more with spring-graphql  as more features become available. We don't know what would be added in the future, but it would be good to have the option. As an example, after experimenting with the `GraphQlTester` (a similar offering to our `DgsQueryExecutor`),   it would be really nice to  allow users to take advantage of this feature ,since it does offer more options for testing. Already, this means we are opening up more of Spring GraphQl's functionality to be exposed more directly to users (not masked as DGS functionality).

In the long term, it also seems reasonable to work towards making all the DGS specific features  work for Spring GraphQl data fetchers and data loaders as well. This can be achieved either by asking Spring folks to offer similar hooks for e.g. data loader instrumentations, or we fill in the gaps for such features. Note that this is not required for the first release, but I can see that being future work for the next phase. 

So with that in mind, it may be beneficial to not disable spring-graphql data fetchers etc. but explicitly state that not everything will  work and that it is not supported.  Thoughts?

Reply:
Actually, I think it is worth pointing out that the main area that needs to be worked out is actually are the data loader concepts - metrics, additional instrumentation etc. The rest of the stuff might be less of a problem. So if we are moving towards that goal of being able to mix both styles of data fetchers/data loaders, this might be the main thing to focus on for the future.